src/operations.cpp

```diff
@@ -94,9 +94,18 @@
 #endif
 
 # ifdef BOOST_POSIX_API
-
 #   include <sys/types.h>
-#   include <sys/stat.h>
+#   if defined(__APPLE__)
+// The stat64() system calls first appeared in Mac OS X 10.5 (Leopard)
+#     if (defined(__MAC_OS_X_VERSION_MIN_REQUIRED) && __MAC_OS_X_VERSION_MIN_REQUIRED >= 1060)
+#       define _DARWIN_USE_64_BIT_INODE
```

Comment:
This macro must be defined before any headers are included. Any header may potentially include `sys/stat.h`, directly or indirectly.